tokio/src/time/sleep.rs

```diff
@@ -226,10 +227,11 @@ pin_project! {
     #[must_use = "futures do nothing unless you `.await` or poll them"]
     pub struct Sleep {
         inner: Inner,
-
+        deadline : Instant,
+        handle: scheduler::Handle,
         // The link between the `Sleep` instance and the timer that drives it.
         #[pin]
-        entry: TimerEntry,
+        entry: Option<TimerEntry>,
     }
```

Comment:
How does this change the size of `Sleep`? Could `TimerEntry` be changed to reduce the change?

Reply:
I think this should be possible. Your idea looks better, I'll give it a try.

Reply:
I put `deadline` and `handle` in `Sleep` here for do not init the `TimerEntry`, but I have to put them to a place to save them temporarily.

Saving `handle` here is to ensure the `timeout` will panic if there is no runtime handle in TLS or we do not `enable_time` or `enable_all`.

The size of `Sleep` is an issue worth considering. Is there any way to solve this problem?

Reply:
In my test, it can improve the performance by reduicng lock contention like lazy initialization when we always let timeout register into the driver. But, on the other hand, if the timeout never register into the driver, this can not bring significant performance improvement.

Reply:
That said, maybe we can use `inner: StdUnsafeCell<Option<TimerShared>>` for this purpose, the size of `Sleep` should be barely changed.

Avoiding `clear_entry` for unregistered `timeout`s is not necessary. In this case, we should avoid constructing the intrusive linked list item.